tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/jwt/JwtTokenRoutes.scala

```diff
@@ -0,0 +1,154 @@
+package com.linagora.tmail.james.jmap.jwt
+
+import com.google.inject.multibindings.Multibinder
+import com.google.inject.name.Named
+import com.google.inject.{AbstractModule, Provides, Singleton}
+import com.linagora.tmail.james.jmap.json.JwtTokenSerializer
+import com.linagora.tmail.james.jmap.jwt.JwtTokenRoutes.{ENDPOINT, TOKEN_DURATION_DEFAULT}
+import com.linagora.tmail.james.jmap.longlivedtoken.{LongLivedTokenAuthenticationStrategy, LongLivedTokenInjectKeys}
+import io.netty.handler.codec.http.HttpHeaderNames.CONTENT_TYPE
+import io.netty.handler.codec.http.HttpResponseStatus.{BAD_REQUEST, FORBIDDEN, INTERNAL_SERVER_ERROR, UNAUTHORIZED}
+import io.netty.handler.codec.http.{HttpMethod, HttpResponseStatus}
+import org.apache.james.jmap.HttpConstants.JSON_CONTENT_TYPE
+import org.apache.james.jmap.core.ProblemDetails
+import org.apache.james.jmap.exceptions.UnauthorizedException
+import org.apache.james.jmap.http.{AuthenticationStrategy, Authenticator}
+import org.apache.james.jmap.json.ResponseSerializer
+import org.apache.james.jmap.routes.ForbiddenException
+import org.apache.james.jmap.{Endpoint, JMAPRoute, JMAPRoutes}
+import org.apache.james.metrics.api.MetricFactory
+import play.api.libs.json.Json
+import reactor.core.scala.publisher.SMono
+import reactor.core.scheduler.Schedulers
+import reactor.netty.http.server.{HttpServerRequest, HttpServerResponse}
+
+import java.nio.charset.StandardCharsets
+import java.time.{Clock, Duration, ZonedDateTime}
+import java.util.stream
+import javax.inject.Inject
+
+case class JwtTokenRoutesModule() extends AbstractModule {
+  override def configure(): Unit = {
+    Multibinder.newSetBinder(binder, classOf[JMAPRoutes])
+      .addBinding().to(classOf[JwtTokenRoutes])
+
+    Multibinder.newSetBinder(binder, classOf[AuthenticationStrategy])
+      .addBinding().to(classOf[LongLivedTokenAuthenticationStrategy])
```

Comment:
This looks very suspicious to me. Please add a test ensuring I cannot call the main JMAP API with a long lived token.